tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -60,18 +59,55 @@ def handle_exceptions(self, exceptions):
         the stack trace"""
         raise exceptions[0]
 
-    def process_in_parallel(self, resources):
+    def process_in_parallel(self, resources, event):
         return ThreadHelper.execute_in_parallel(
             resources=resources,
-            execution_method=self.process_resource_set,
+            event=event,
+            execution_method=self._process_resources,
             executor_factory=self.executor_factory,
             log=self.log,
             max_workers=self.max_workers,
             chunk_size=self.chunk_size
         )
 
+    def _process_resources(self, resources, event):
+        self._prepare_processing()
+
+        for r in resources:
+            try:
+                self._process_resource(r)
+            except CloudError as e:
+                TagHelper.log.error("Failed to process resource.\n"
```

Comment:
I know it's a WIP but just in case you didn't notice - not sure you mean to use the TageHelper log here?

Reply:
this is why [wip] :)

Reply:
Alright I'll wait :D